### PR TITLE
Use plugins.providers rather than setting PATH

### PR DESCRIPTION
**Pull request by Frassle**
# Description

This changes the integration tests to use the new plugin link functionality rather than setting PATH.

Also a small change to the `DebugLogging` option to also set "--logflow" as it was really awkward debugging without plugin logs.

There was a `LocalDependency` option already on `ProgramTestOptions` but it was unused so I've renamed it to `LocalProviders` to be more clear that this is for setting local providers, and if we want to support other plugin types we'll need new options added.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/testing/integration/program.go [hunk 1]
```diff
@@ -560,9 +560,6 @@ func (opts ProgramTestOptions) With(overrides ProgramTestOptions) ProgramTestOpt
 	if overrides.PrepareProject != nil {
 		opts.PrepareProject = overrides.PrepareProject
 	}
-	if overrides.LocalDependencies != nil {
-		opts.LocalDependencies = append(opts.LocalDependencies, overrides.LocalDependencies...)
-	}
```
Comment (justinvp) [inline pkg/testing/integration/program.go:565]: We should keep this for the renamed `LocalProviders` option.

File: pkg/testing/integration/program.go [hunk 2]
```diff
@@ -311,8 +311,8 @@ type ProgramTestOptions struct {
 	// uses Node, Python, .NET or Go.
 	PrepareProject func(*engine.Projinfo) error
 
-	// Array of dependencies which come from local packages.
-	LocalDependencies []LocalDependency
+	// Array of provider plugin dependencies which come from local packages.
+	LocalProviders []LocalDependency
```
Comment (AaronFriel) [inline pkg/testing/integration/program.go:315]: Code Search suggests this is an OK change 👍 

https://cs.github.com/?scope=org%3Apulumi&scopeName=pulumi&q=LocalDependencies

File: pkg/testing/integration/program.go [hunk 3]
```diff
@@ -560,9 +560,6 @@ func (opts ProgramTestOptions) With(overrides ProgramTestOptions) ProgramTestOpt
 	if overrides.PrepareProject != nil {
 		opts.PrepareProject = overrides.PrepareProject
 	}
-	if overrides.LocalDependencies != nil {
```
Comment (AaronFriel) [inline pkg/testing/integration/program.go:563]: Any reason for removing this from the "overrides" options?

**Review by AaronFriel — APPROVED** [review]
Changes lgtm.

CC @t0yv0 that I will want to rebase #10624 on this to address merge conflicts. Will wait for your review to complete.

**Review by justinvp — APPROVED** [review]
LGTM

File: pkg/testing/integration/program.go (hunk 2 above)
Comment (justinvp) [reply pkg/testing/integration/program.go:315]: Yeah, I was going to say, that Fraser added this very recently, so I doubt anyone (even outside of Pulumi) has used it yet. So I'm good with the rename.

File: pkg/testing/integration/program.go (hunk 3 above)
Comment (Frassle) [reply pkg/testing/integration/program.go:563]: Oh nope that was a mistake will readd it, good catch

**Comment by Frassle** [comment]
bors merge

**Comment by Frassle** [comment]
bors retry

**Comment by AaronFriel** [comment]
bors cancel

**Comment by Frassle** [comment]
Bors merge

**Comment by Frassle** [comment]
bors merge

**Pull request merged**